### PR TITLE
Pure Swift Version

### PR DESCRIPTION
**Pull request by adamkaplan**
This is a full rewrite of TDOAuth in pure Swift.

By default, the Pod includes the subspec `TDOAuth/ObjC` which provides a backwards compatible Objective-C API. This subspec exposes _the TDOAuth.h header_ from the old Objective-C version to guarantee compatibility. (Behind the scenes in `TDOAuth.m`, there is a shim into the new Swift version, which consists mostly of copied old code).

The original Objective-C unit tests have been included (copy-paste) as a compatibility check. In addition, comprehensive and independent unit tests of the new Swift version are also in place.

Swift compatibility includes 4.0, 4.2, 5.0

To use the pure Swift version _without any Objective-C code_, use the `TDOAuth/Swift` subspec.

`
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.
`

### CODE REVIEW

File: Example/Tests/OAuth1Spec.swift [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        let expected = "OAuth oauth_nonce=\"kllo9940pd9333jh\", oauth_signature_method=\"HMAC-SHA1\", oauth_consumer_key=\"dpf43f3p2l4k3l03\", oauth_timestamp=\"1191242096\", oauth_version=\"1.0\", oauth_signature=\"Jg5MXVnexhzMDTv7IBUy3goIGqc%3D\""
+
+        XCTAssertNotNil(signedRequest)
+        XCTAssertEqual(authHeader, expected)
+    }
+
+            // This is a nonsensical edge case wherein the access_token is optional
+            // but the RFC does not state that the token_secret MUST be omitted
+            // if the access_token is not provided. Logically the client wouldn't
+            // have a secret without a token, but whatever...
```
Comment (plasmabal) [inline Example/Tests/OAuth1Spec.swift]: This block of comments is out of indentation. :)

**Comment by adamkaplan** [comment]
> I think public classes should have prefix `TDO` in their names to avoid possible collisions with other pods that may be used by the host app. Therefore names like `OAuth1` and `HMACSigner` should be renamed to `TDOAuth1` and `TDOHMACSigner`.
> 
> And the code is structured beautifully! I enjoyed reading the code. 👍

That's true definitely for Objective-C, but for Swift I don't think it's needed anymore. Similar to how Apple renamed `NSURLSession` to `URLSession`. Basically, the conflict is resolved by name spacing of the module, i.e. `import TDOAuth`.

So if there is a conflict with use of `OAuth1`, the compiler will show an error and the user should resolve by specifying `TDOAuth.OAuth1`. Normally the compiler will use type inference to add the `TDOAuth.` part under the hood.

File: Example/Tests/OAuth1Spec.swift (hunk 1 above)
Comment (adamkaplan) [reply Example/Tests/OAuth1Spec.swift]: Fixed

**Comment by adamkaplan** [comment]
> I'm not sure if I can approve. Anyway...

Yeah the repo is restricted by Yahoo OSWG, but official ✅ doesn't matter – thanks for the review!

**Pull request merged**